generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/KTypeInfo.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.generator.internal.types
+
+/**
+ * Internal metadata class we can use to forward info about the type we are generating.
+ * If there is no metadata to add, create the class with default values.
+ */
+internal data class KTypeInfo(
```

Comment:
just wondering whether this should also hold the actual type information

Reply:
We could, it would just require a lot more code changes. I figured it was more clear if we kept the `KType` argument separate from the custom class we had for more metadata about how and where the `KType` was used

Reply:
maybe rename it to `KTypeMetadata` then? also is it possible to create an instance without a field name (currently it defaults to null)?

Reply:
I updated the type name to `GraphQLKTypeMetadata`. And yes it is possible to create the class without a field name in the use of additional types, ie you may never return an interface implementation explicitly in a field